### ISSUE TITLE
CypressError when using cy.server()

### ISSUE THREAD

**Issue by lorennorman** [cypress-io/cypress#12]
My code:
```
describe 'Navigating there [00h]', ->
  it 'is in the menu [00i]', ->
    cy.server().route 'POST', /users.json/, {}
```
...crashes immediately with:
```
CypressError: sinon.js was not found in the remote iframe's window.  This may happen if you testing a page you did not directly cy.visit(..).  This happens when you click a regular link.
```
...and it makes me very :crying_cat_face:

**Comment by brian-mann** [comment]
Have you loaded your app via `cy.visit` before hand?

That's what the error is saying.  Sinon is dynamically injected into your page after you `cy.visit` it.

**Comment by lorennorman** [comment]
Interesting. No I didn't load it first because it calls the server
immediately and I want it mocked first. Trying it now...

**Comment by lorennorman** [comment]
Yeah it works now. I'll report a typo in the error message instead: 
`if you testing a page`

:+1:

**Comment by brian-mann** [comment]
Yah that error message needs work.

**Comment by brian-mann** [comment]
I think this does bring up an interesting point.

The way `visit` works is that it doesn't resolve until your page fires the `load` event.  By the time the page does that, it's possible you had scripts that immediately made AJAX requests when they first loaded.

If that's the case, those requests would go out, since they'd happen before `visit` resolves, and therefore before the `server` command was even processed.

I'll need to set up a scenario where I can cause this to happen.  `visit` does take some options like `onLoad` and `onBeforeLoad`, but that wouldn't help starting a server because that's async anyway.

I may have to expand `visit` to allow for immediately starting a server prior to any scripts being parsed or loaded.

I think the situation works in your case because you're either not firing anything immediately on load, or because you're using angular, it doesn't fire off the requests until much later.  Though its possible due to a timing issue sometimes it may request faster than `server` can be booted.

When are your requests being made?

**Comment by lorennorman** [comment]
We're using UI Router and there are requests in the resolve block. I
believe that means that Angular loads fully and fires the event, then it
starts the navigation cycle which leads to the request being made. I really
don't know what I'm talking about, though.

Maybe you could make .server() hold its state until the appropriate time
then slip it in in the case where visit() hasn't been called yet? Even if
not, maybe you can better detect this situation and improve the error
message, "you called .server() before you called .visit()", because when
you start mentioning sinon and iframes you break the magic barrier and I
have to understand the entire system suddenly.

Instead of accepting that I was wrong, I assumed something was broken.
Because I'm an asshole. And so is everyone else.

**Comment by brian-mann** [comment]
Such wise words.

Yah I'll make some adjustments for sure.

**Comment by brian-mann** [comment]
This is fixed in `0.4.3`. [Changelog](https://github.com/cypress-io/cypress/wiki/Changelog#043-04092015)
